src/result.zig

```diff
@@ -0,0 +1,11 @@
+pub fn Result(comptime T: type, comptime E: type) type {
```

Comment:
Maybe we should combine Maybe and Result and then Maybe is Result with a Mixin applied?